src/google/protobuf/util/internal/default_value_objectwriter.cc

```diff
@@ -156,8 +153,8 @@ DefaultValueObjectWriter* DefaultValueObjectWriter::RenderString(
   } else {
     // Since StringPiece is essentially a pointer, takes a copy of "value" to
     // avoid ownership issues.
-    string_values_.push_back(new string(value));
-    RenderDataPiece(name, DataPiece(*string_values_.back(), true));
+    string_values_.emplace_back(value);
+    RenderDataPiece(name, DataPiece(string_values_.back(), true));
```

Comment:
Sorry to keep going back and forth but I realized that I told you the wrong thing before. It turns out that `DataPiece` hangs onto a reference to the string data here instead of making a copy. That means that `string_values_` can't be a `vector<string>` because when the vector resizes it will move elements around and create dangling references.

I think we should go back to your original idea of using `std::vector<std::unique_ptr<std::string>>`. We have to avoid `std::make_unique` since we don't have that available in C++11, but something like `string_values_.emplace_back(new string(value));` should work if I'm not mistaken.

Reply:
Ok, no problem to revert to original pull request's implementation=)
(guess in this time, everything going well, especially unit tests)